### PR TITLE
[parse] Update cloud code definitions for Parse server 3.X

### PR DESCRIPTION
**Pull request by AlexandreHetu**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://parseplatform.org/parse-server/api/master/Parse.Cloud.html
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

The cloud code definitions in the JS SDK project do not correspond to the actual code in Parse-Server. Since Parse Server 3.X, only Promises or raw returns are supported by cloud functions. This PR makes sure that the definitions match the Parse Server cloud code documentation.

### CODE REVIEW

**Review by agoldis — COMMENTED** [review]
Nice job, @AlexandreHetu  👍🏻 - looks good!

Would you mind please modifying the header to

```
// Type definitions for parse 2.1.0 and parse-server <version>
```

and also, add test cases for 

- `afterFind`
- `define`
- `job`

Since you've touched those.

**Comment by AlexandreHetu** [comment]
Hi there! Thanks for the feedback. I ended up also fixing the `JobRequest` interface.

**Review by agoldis — APPROVED** [review]
👍🏻

File: types/parse/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for parse 2.1.0
+// Type definitions for parse 2.1.0 and parse-server 3.1.3
```
Comment (rbuckton) [inline types/parse/index.d.ts]:
```suggestion
// Type definitions for parse 2.1.0
```
The type definitions comment line should only include the package and version of the actual package. If `parse-server` has a dependency on `parse`, that should be reflected in the type definitions for `parse-server`.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/parse/index.d.ts (hunk 1 above)
Comment (agoldis) [reply types/parse/index.d.ts]: @AlexandreHetu let's just remove the comment, looks like it would be most effective approach

**Comment by AlexandreHetu** [comment]
The comment has been updated. @rbuckton, could you take a look!?

Thanks!

**Pull request merged**